xl2times/utils.py

```diff
@@ -60,7 +60,14 @@ def explode(df, data_columns):
     :return:                Tuple with the exploded dataframe and a Series of the original
                             column name for each value in each new row.
     """
-    data = df[data_columns].values.tolist()
+    # Handle duplicate columns (https://pandas.pydata.org/docs/user_guide/duplicates.html)
+    if len(set(data_columns)) < len(data_columns):
+        cols = df.columns.to_list()
+        data_cols_idx = [idx for idx, val in enumerate(cols) if val in data_columns]
+        data = df.iloc[:, data_cols_idx].values.tolist()
+    else:
+        data = df[data_columns].values.tolist()
```

Comment:
This works for austimes, but it'll always just discard all but the _first_ column of any set of columns with duplicate names.
Whereas I understood that we want to want to 'fill-right' their values (e.g. using the merge function from #198) - so that any missing values in the right-most column get filled with non-missing values from the next column to the left, etc.

Suggest we replace this with:
```python
if len(set(data_columns)) < len(data_columns):
    df =  _merge_duplicate_named_columns(df)
```

Reply:
I can merge this in here and test it if you like @olejandro ?

Reply:
@SamRWest, as I've tried to explain in #198, there may be an attribute alias, e.g.:

| PROCESS | NCAP_START| START | NCAP_START |
| --- | --- | --- | --- |
| PRC1 | 2012|          | 2020 |
| PRC2 | 2012| 2016 |          |
| PRC3 |         | 2016 | 2020 |

Using your approach will turn this to: 

| PROCESS | START | NCAP_START |
| --- | --- | --- |
| PRC1 |          | 2020 |
| PRC2 | 2016 | 2012 |
| PRC3 | 2016 | 2020 |

It will later be transformed into :

| PROCESS | ATTRIBUTE | VALUE |
| --- | --- | --- |
| PRC1 | NCAP_START | 2020 |
| PRC2 | NCAP_START | 2016 |
| PRC2 | NCAP_START | 2012 |
| PRC3 | NCAP_START | 2016 |
| PRC3 | NCAP_START | 2020 |

Resulting in an incorrect `NCAP_START` value for `PRC2`, because the rows below overwrite the rows above.

What I try to ensure with my changes is that the original table results in:

 | PROCESS | ATTRIBUTE | VALUE |
| --- | --- | --- |
| PRC1 | NCAP_START | 2012 |
| PRC1 | NCAP_START | 2020 |
| PRC2 | NCAP_START | 2012 |
| PRC2 | NCAP_START | 2016 |
| PRC3 | NCAP_START | 2016 |
| PRC3 | NCAP_START | 2020 |

This way the  `NCAP_START` value for `PRC2` will be correct.

Reply:
I think the approach proposed in #198 should be applied in `normalize_column_aliases` and only apply to columns not in  `config.all_attributes` and `config.attr_aliases` sets.

Reply:
> This happens to be ok in this particular example because the values are identical, but if, say, the second `fixom` value was `None`, then dropping first duplicate rows would give the wrong answer later.

@SamRWest, I've modified Demo 7 to cover this case. It seems to be working fine.

Reply:
Oh ok, I think I get it (didn't realise how aliases worked before).  The explode() and later dropping of duplicate rows kind of does the work or resolving duplicates for you.

If you have time, it'd probably be worth writing a unit test to spell this logic out.  But if you've checked that it matches VEDA, then that's the main thing, so I'm happy.

Reply:
Ok, sure! Tbh, I have never written one, but there is always the first time. 😃

Reply:
Nothing to it, just make a fake dataframe (like your examples above) run it through the appropriate transforms, and `assert` what it should look like on the other end.
All the boilerplate is already set up in `test_transforms.py`, so just make a new `test_whatever()` function in there, and it'll get checked in every CI run so we don't accidentally change the desired behaviour later.